### PR TITLE
AddExactCoverWithGlobalElements( SkeletalFinSets, ... )

### PR DESCRIPTION
**Pull request by mohamed-barakat**
needs Toposes v2022.11-03

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
See the above-referenced `FunctorCategories`-PR for the application.

File: gap/ToolsDerivedMethods.gi [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          [ TerminalObject, 1 ] ],
+        
+  function ( cat, A )
+    
+    return MorphismsOfExternalHom( cat, TerminalObject( cat ), A );
+    
+end :
+  CategoryFilter := cat ->
+                    IsCategoryOfFinSets( cat ) or
+                    IsCategoryOfSkeletalFinSets( cat ) );
```
Comment (zickgraf) [inline gap/ToolsDerivedMethods.gi:19]: If a "derivation" matches only a finite number of concrete primitive categories, we usually simply install it primitively. So I suggest doing that in this case, too.

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: gap/ToolsDerivedMethods.gi (hunk 1 above)
Comment (mohamed-barakat) [reply gap/ToolsDerivedMethods.gi]: Done.

File: gap/precompiled_categories/CategoryOfSkeletalFinSetsPrecompiled.gi [hunk 2]
```diff
@@ -304,6 +304,27 @@ end
...
+function ( cat_1, arg2_1 )
+    local hoisted_1_1, hoisted_2_1, hoisted_3_1, deduped_4_1, deduped_5_1;
+    deduped_5_1 := Length( arg2_1 );
+    deduped_4_1 := [ 0 .. deduped_5_1 - 1 ];
+    hoisted_3_1 := CreateCapCategoryObjectWithAttributes( cat_1, Length, 1 );
+    hoisted_2_1 := deduped_5_1;
+    hoisted_1_1 := deduped_4_1;
+    return List( deduped_4_1, function ( i_2 )
+            local deduped_1_2;
+            deduped_1_2 := CAP_JIT_INCOMPLETE_LOGIC( hoisted_1_1[1 + i_2] );
```
Comment (mohamed-barakat) [inline gap/precompiled_categories/CategoryOfSkeletalFinSetsPrecompiled.gi:320]: For the already existing logic template

```
CapJitAddLogicTemplate(
    rec(
        variable_names := [ "pos_end", "index" ],
        variable_filters := [ IsInt, IsInt ],
        src_template := "[ 0 .. pos_end ][index]",
        dst_template := "index - 1",
    )
);
```

it is probably too late. Right?

Comment (zickgraf) [reply gap/precompiled_categories/CategoryOfSkeletalFinSetsPrecompiled.gi:320]: Correct, this occurs all over the place.

**Review by zickgraf — APPROVED** [review]

**Pull request merged**